test/e2e/storage/sanity.go

```diff
@@ -65,6 +65,7 @@ var _ = Describe("sanity", func() {
 		// and deletes all extra entries that it does not know about.
 		TargetPath:  "/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pmem-sanity-target.XXXXXX",
 		StagingPath: "/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pmem-sanity-staging.XXXXXX",
+		IDGen: &sanity.DefaultIDGenerator{},
```

Comment:
This shouldn't be necessary. I think it's a bug in csi-test, which we should fix there: https://github.com/kubernetes-csi/csi-test/blob/82b05190c167c52bb6d5aaf2e1d7c833fa539783/pkg/sanity/sanity.go#L161-L163 was added to `Test`, but not `https://github.com/kubernetes-csi/csi-test/blob/82b05190c167c52bb6d5aaf2e1d7c833fa539783/pkg/sanity/sanity.go#L183`.

We should have a common function that is used by both code paths to sanitize the config.

@okartau as you found this, can you do the upstream fix?

In this repo we can keep `IDGen: &sanity.DefaultIDGenerator{}` as a temporary workaround for 1.2.2, but there should be a `TODO` comment above it about removing it.

Reply:
https://github.com/kubernetes-csi/csi-test/pull/220